test/extended/builds/new_app.go

```diff
@@ -0,0 +1,41 @@
+package builds
+
+import (
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[builds][Slow] new-app with long name", func() {
```

Comment:
does this really need to be an extended test?  how about adding a test here instead:
https://github.com/openshift/origin/blob/master/test/cmd/newapp.sh